### PR TITLE
Historical notes in the docs

### LINKED ISSUE #2512: Historical notes in the documentation
On our docs it is common to have historical notes explaining how something was done in previous versions and is no longer recommended or has been deprecated. 

For example in [the fixture docs](https://docs.pytest.org/en/latest/fixture.html?highlight=addfinalizer#fixtures-as-function-arguments):

> In versions prior to 2.3 there was no ``@pytest.fixture`` marker
    and you had to use a magic ``pytest_funcarg__NAME`` prefix
    for the fixture factory.  This remains and will remain supported
    but is not anymore advertised as the primary means of declaring fixture
    functions.

While these blurbs are very useful for users looking for them in the docs after seeing them in old code, I think it is distracting for first time users.

How about:

1. Move all historical notes to the bottom of their files?
2. Move all historical notes to a separate file?

I like `2` best because it won't distract new readers at all and are still searchable by the users who are interested in them.

Opinions?

@hackebrot @pfctdayelise

### PR DESCRIPTION
**Pull request by nicoddemus**
Fixes #2512

Also:

* Replaced deprecated `config.option.<name>` usages
* Fixed some contact information

### CODE REVIEW

File: doc/en/unittest.rst [hunk 1]
```diff
@@ -191,12 +191,12 @@ was executed ahead of the ``test_method``.
...
-   grows a plugin system that should help as well.  In the meanwhile, the 
-   above ``usefixtures`` and ``autouse`` examples should help to mix in 
-   pytest fixtures into unittest suites.  And of course you can also start
-   to selectively leave away the ``unittest.TestCase`` subclassing, use
-   plain asserts and get the unlimited pytest feature set.
+
+   The above ``usefixtures`` and ``autouse`` examples should help to mix in
+   pytest fixtures into unittest suites.
+
+   You can also consider incrementally leave away ``unittest.TestCase`` subclassing,
```
Comment (hackebrot) [inline doc/en/unittest.rst]: > You can also gradually move away from subclassing from ``unittest.TestCase`` to *plain asserts* and benefit from the full pytest feature set.

**Comment by nicoddemus** [comment]
Thanks @hackebrot, applied all your requested changes! 👍

File: doc/en/unittest.rst [hunk 2]
```diff
@@ -191,12 +191,12 @@ was executed ahead of the ``test_method``.
...
-   above ``usefixtures`` and ``autouse`` examples should help to mix in 
-   pytest fixtures into unittest suites.  And of course you can also start
-   to selectively leave away the ``unittest.TestCase`` subclassing, use
-   plain asserts and get the unlimited pytest feature set.
+
+   The above ``usefixtures`` and ``autouse`` examples should help to mix in
+   pytest fixtures into unittest suites.
+
+   You can also gradually move away from subclassing from ``unittest.TestCase``
+   to *plain asserts* and the benefits from the full pytest feature set.
```
Comment (hackebrot) [inline doc/en/unittest.rst]: You can also gradually move away from subclassing from ``unittest.TestCase`` to *plain asserts* and then start to benefit from the full pytest feature set step by step.


Hmm is this better? 🤔

Comment (nicoddemus) [inline doc/en/unittest.rst]: Done, indeed it is better!

**Pull request merged**